lib/collection/src/collection_state.rs

```diff
@@ -64,9 +89,16 @@ impl State {
             let mut shards_holder = collection.shards_holder.write().await;
             match shards_holder.get_shard(&shard_id) {
                 Some(shard) => {
-                    if shard.peer_id(this_peer_id) != peer_id {
-                        // shard registered on a different peer
-                        log::warn!("Shard movement between peers is not yet implemented. Failed to move shard {shard_id} to peer {peer_id}")
+                    let old_peer_id = shard.peer_id(this_peer_id);
+                    // shard was moved to a different peer
+                    if old_peer_id != peer_id {
+                        collection
+                            .finish_shard_transfer(ShardTransfer {
```

Comment:
Why is it `finish_shard_transfer` and not `abort_shard_transfer` in that case?

Reply:
In this case shard transfer was already finished - shard is moved as far as network knows - it has a new peer id. So the only thing left on this peer is to try to finish it here. Which means that it will update all the appropriate references in remote shards and promote if there were any temp or proxy shards.